### ISSUE TITLE
Some filters not working in v0.48.x

### ISSUE THREAD

**Issue by Dialeee** [metabase/metabase#37623]
### Describe the bug

When two or more filters are pointing to the same variable in the card of a dashboard the filter won't do anything.
This didn't happened in previous versions.

Even if one of the filters is empty it won't filter.

### To Reproduce

1. Create a dahboard with two related filters and a card with a variable in postgresql code
2. Link both filters to the card's variable
3. Try to filter 

### Expected behavior

It should filter using only both filters' equal values like in the previous versions

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "es-ES",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.21+9",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.21",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.21+9",
    "os.name": "Linux",
    "os.version": "5.4.0-163-generic",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "15.3 (Debian 15.3-1.pgdg120+1)"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.6.0"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-01-05",
      "tag": "v0.48.2",
      "hash": "e66c075"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

After updating to v0.48 most of our dashbord's filters stopped working because of this

**Comment by qnkhuat** [comment]
Hey @Dialeee, I can't reproduce this.
Here is what I did
1. create a native question with this query `select * from products where category = {{category}} and rating = {{rating}};` where category is a text variable and rating is a number variable
2. added the above question to a dashboard, create 2 filters, one text, one number and link to the variables from the card
3. Input "Doohickey" for the text filter and 4 for the number filter, and here is the result I have
![Screenshot 2024-01-12 at 16 35 33](https://github.com/metabase/metabase/assets/25661381/3e71d11c-da5a-4ea5-b99a-fbc375eb625f)
it seems fine to me.

anything wrong with my reproduction steps? also it'd be best if you provide the reproduction steps using the sample database, it'll be easier for us to debug

**Comment by Dialeee** [comment]
> Hey @Dialeee, I can't reproduce this. Here is what I did
> 
> 1. create a native question with this query `select * from products where category = {{category}} and rating = {{rating}};` where category is a text variable and rating is a number variable
> 2. added the above question to a dashboard, create 2 filters, one text, one number and link to the variables from the card
> 3. Input "Doohickey" for the text filter and 4 for the number filter, and here is the result I have
>    ![Screenshot 2024-01-12 at 16 35 33](https://private-user-images.githubusercontent.com/25661381/296193509-3e71d11c-da5a-4ea5-b99a-fbc375eb625f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUwNTMxMzQsIm5iZiI6MTcwNTA1MjgzNCwicGF0aCI6Ii8yNTY2MTM4MS8yOTYxOTM1MDktM2U3MWQxMWMtZGE1YS00ZWE1LWI5OWEtZmJjMzc1ZWI2MjVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEyVDA5NDcxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJkN2M3YzM5NDU1MTM3NzQyNTllMTMxYmVlNDk0ODJhZmM1N2Y0MGFjOWU3ODc1NTNkODVmMjY2MjUyYzMxNGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cbThJlDDx1seG3sZX_DzMt8osgEeMSKnYPZlabGjWas)
>    it seems fine to me.
> 
> anything wrong with my reproduction steps? also it'd be best if you provide the reproduction steps using the sample database, it'll be easier for us to debug

I don't have the sample database anymore sorry...

Both filters need to be linked to the same variable for example you could create a query like this:
select * from products [[where {{category}}]];

And create two category filters in the dashboard then link both to the {{category}} variable. Also {{category}} must be a category mapped field so that you can choose multiple values from a list in both filters.

**Comment by paoliniluis** [comment]
Adding the discourse conversation about this https://discourse.metabase.com/t/the-filter-on-the-dashboard-doesnt-work/62628/5

**Comment by bshepherdson** [comment]
This issue is a duplicate of #38012 where SQL questions only filter properly if every filter aimed at a particular variable has a value.

That issue was debugged and fixed, and the fix will be in the 48.4 release scheduled for today.

**Comment by chandu8686** [comment]
Hi Team , Metabase latest version 0.48.X , Postgres SQL Driver 42.6.0 is showing  
Package
 org.postgresql:postgresql (
[Maven](https://github.com/advisories?query=ecosystem%3Amaven)
)
Affected versions
< 42.2.28
< 42.3.9
< 42.4.4
< 42.5.5
< 42.6.1
< 42.7.2
Patched versions
42.2.28, 42.2.28.jre7
42.3.9
42.4.4
42.5.5
42.6.1
42.7.2
How are we going to use the Postgres SQL latets Driver
